src/main/java/de/tum/in/www1/artemis/web/rest/plagiarism/PlagiarismCaseResource.java

```diff
@@ -66,6 +72,31 @@ public ResponseEntity<List<PlagiarismCase>> getPlagiarismCasesForCourseForInstru
         return getPlagiarismCasesResponseEntity(plagiarismCases);
     }
 
+    /**
+     * Retrieves all plagiarism cases related to an exam for the instructor view.
+     *
+     * @param courseId the course id to which the exam belongs
+     * @param examId the id of the exam
+     * @return all plagiarism cases of the course
+     */
+    @GetMapping("courses/{courseId}/exams/{examId}/plagiarism-cases/for-instructor")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    public ResponseEntity<List<PlagiarismCase>> getPlagiarismCasesForExamForInstructor(@PathVariable long courseId, @PathVariable long examId) {
+        log.debug("REST request to get all plagiarism cases for instructor in exam with id: {}", examId);
+        Course course = courseRepository.findByIdElseThrow(courseId);
```

Comment:
You can skip this database call, as this Object seems to be fetched by the other database query as well.

Reply:
That would allow a user who does not have access to the course to get knowledge about whether the plagiarism cases are empty or not. But then they can always try other `examId`s if they are instructors in at least one course so saving the database query may be more important here. Whet do you think?

Reply:
Fair enough. We should not leak if there are plagiarism cases or not. We could do something like this:
```
        var plagiarismCases = plagiarismCaseRepository.findByExamIdWithPlagiarismSubmissionsAndComparison(examId);
        Course course;
        if (!plagiarismCases.isEmpty()) {
            course = plagiarismCase.getExercise().getExerciseGroup().getExam().getCourse();
        } else {
            course = courseRepository.findByIdElseThrow(courseId);
        }
        authenticationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
```
However that is a bit more complicated. Not sure what is better.

Reply:
Then maybe we can keep the current version for simplicity since we are using `findById`, it is highly likely that the relevant course is in Hibernate cache and it won't hit the database (I was able to confirm this locally).